### ISSUE TITLE
[TOPI] Add count_include_pad attr to AvgPool Op

### ISSUE THREAD

**Issue by fumihwh** [dmlc/tvm#1144]
Reference: https://pytorch.org/docs/master/nn.html?highlight=avg_pool2d#torch.nn.AvgPool2d

Current, when calculating average pool, padding will be counted.

But in Caffe2 and Tensorflow, they do average pool without counting padding.
https://github.com/caffe2/caffe2/blob/master/caffe2/operators/pool_op.cc#L556

e.g.:
```
0 0 0 ...
0 1 2 ...
0 3 4 ...
```
Above average pool will be:
without padding: ( 1 + 2 + 3 + 4 ) / 4 
with padding: ( 1 + 2 + 3 + 4 ) / 9 

I think we should add count_include_pad to AvgPool.

**Comment by tqchen** [comment]
This can likely be solved via a select condition to add the tail padding unit calculation.

**Comment by tqchen** [comment]
topi support is added thanks to @nishi-t need to update nnvm to support this

**Comment by tqchen** [comment]
Thanks for @nishi-t's contribution
